### ISSUE TITLE
Mermaid flowcharts render blank boxes when a text line is too long (at ~30 chars)

### ISSUE THREAD

**Issue by tuurep** [jannis-baum/Vivify#195]
[//]: # (Hello and thank you for taking the time to submit a bug report!       )

[//]: # (To make sure we can resolve this as quickly as possible, please       )
[//]: # (carefully fill out the information requested on this template.        )

[//]: # (Hint: Lines that look like these are Markdown comments and won't be   )
[//]: # (      visible on the issue after you submit it so you don't need to   )
[//]: # (      delete them.                                                    )

### Description

When a flowchart node has a line of text that's too long, the whole node renders as a small blank box. Example:

````
```mermaid
flowchart LR
    A["Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."]
```
````

Screenshot:

![image](https://github.com/user-attachments/assets/ccec3e5e-a367-4b94-91cb-f452babd32b2)

With manual wrapping (a newline or a `<br>` tag), it can be made to render if the lines are about 27 characters or less, for example:

````
```mermaid
flowchart LR
    A["Lorem ipsum dolor sit
       amet, consectetur
       adipiscing elit, sed
       do eiusmod tempor
       incididunt ut labore
       et dolore magna
       aliqua."]
```
````

Screenshot:

![image](https://github.com/user-attachments/assets/d010c4bc-2433-4f75-9f34-4239b21ecd9b)

Blank box in comparison to normal boxes with text:

````
```mermaid
flowchart LR
    foo-->lorem["Lorem ipsum dolor sit amet, consectetur."]
    lorem-->bar
```
````

![image](https://github.com/user-attachments/assets/d6770cab-a246-4fe9-b9dd-a19f7aa528f1)

### Expectation

This blog post reveals 2 things:

https://docs.mermaidchart.com/blog/posts/automatic-text-wrapping-in-flowcharts-is-here

1. Long lines should render as long boxes to begin with.
2. Hard wrapping should not be necessary if the string is formatted in this way:

    - ``"`Lorem ipsum dolor sit amet, consectetur.`"``
      - (notice: backticks)

  .. in which case it should auto wrap (i.e. soft wrap), but in Vivify's rendering this isn't working.

### Context

- Firefox 133.0
- Arch Linux
- [`vivify-git`](https://aur.archlinux.org/packages/vivify-git) (AUR)
  - `vivify-server v0.6.3.r0.g5d07d16-1-aur`

**Comment by tuurep** [comment]
Yes, works perfect on Chromium's side, and those issues are related.

A potential workaround for Firefox is setting [`htmlLabels: false`](https://mermaid.js.org/config/schema-docs/config-defs-flowchart-diagram-config-properties-htmllabels.html)

Looking around trying to understand what `htmlLabels` is, I find this:

> Some diagrams have a `htmlLabels: false` config setting that you can pass that might stop them from using `<foreignObject>`/embedded HTML.
>
> -- https://github.com/mermaid-js/mermaid-cli/issues/691#issuecomment-2413939632

Examples:

````
```mermaid
%%{init: {"flowchart": {"htmlLabels": false}} }%%

flowchart LR
    A["Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."]
```
````

![image](https://github.com/user-attachments/assets/bf3d314a-682c-4f6f-a549-3b9528d1356c)

````
```mermaid
%%{init: {"flowchart": {"htmlLabels": false}} }%%

flowchart LR
    foo-->lorem["Lorem ipsum dolor sit amet, consectetur."]
    lorem-->bar
```
````

![image](https://github.com/user-attachments/assets/ff039be1-7a70-45b6-939a-2432606d4f0d)

But that probably has some undesirable effects as well, what I spotted right now was this:

````
```mermaid
flowchart LR
    a -- hello world --> b
```

```mermaid
%%{init: {"flowchart": {"htmlLabels": false}} }%%

flowchart LR
    a -- hello world --> b
```
````

![image](https://github.com/user-attachments/assets/f649e7db-de2f-4e2c-89ec-a86dce0ef860)

**Comment by jannis-baum** [comment]
So do you think there is anything we can/should do on Vivify's side? If not I'd close this

**Comment by tuurep** [comment]
Yes, I made sure to subscribe to the issues in mermaid repos - really depends on those.
